src/java/org/apache/cassandra/db/compaction/CompactionIterator.java

```diff
@@ -206,6 +206,17 @@ private UnfilteredPartitionIterators.MergeListener listener()
     {
         return new UnfilteredPartitionIterators.MergeListener()
         {
+            private boolean rowProcessingNeeded()
+            {
+                return type == OperationType.COMPACTION && controller.cfs.indexManager.hasIndexes();
```

Comment:
```suggestion
                return type == OperationType.COMPACTION && controller.cfs.indexManager.handles(IndexTransaction.Type.COMPACTION);
```

Reply:
This will trivially resolve to `true` right now unless we're using SAI, which doesn't care. (see `StorageAttachedIndex#indexerFor()`)